src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        head_mask (`torch.FloatTensor` of shape `(num_heads,)` or `(num_layers, num_heads)`, *optional*):
+            Mask to nullify selected heads of the self-attention modules. Mask values selected in `[0, 1]`:
+        inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
+            Optionally, instead of passing `input_ids` you can choose to directly pass an embedded representation. This
+            is useful if you want more control over how to convert `input_ids` indices into associated vectors than the
+            model's internal embedding lookup matrix.
+        decoder_inputs_embeds (`torch.FloatTensor` of shape `(batch_size, target_sequence_length, hidden_size)`, *optional*):
+            Optionally, instead of passing `decoder_input_ids` you can choose to directly pass an embedded
+            representation. If `past_key_values` is used, optionally only the last `decoder_inputs_embeds` have to be
+            input (see `past_key_values`). This is useful if you want more control over how to convert
+            `decoder_input_ids` indices into associated vectors than the model's internal embedding lookup matrix.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+def make_attention_mask_torch(total_seq, output_seq, input_len):
+    device = input_len.device
+    i = torch.arange(total_seq)[:, None].to(device)
+    j = torch.arange(output_seq).to(device)
+    m = i >= j - output_seq + total_seq
+    lm_mask = m.float()  # language model mask
+    lm_mask = torch.reshape(lm_mask, [total_seq, output_seq])
+    # lm_mask shuld be [sequence, sequence]
+    weight = torch.transpose(torch.arange(output_seq)[:, None].to(device) < input_len, 1, 0)
+    weight = weight.float()  # Masked language model mask
+    # weight shuld be [batch, sequence]
+    mlm_mask = torch.reshape(weight, [1, -1, output_seq]).float()
+    mlm_ones = torch.ones(size=[total_seq, 1, 1], dtype=torch.float32).to(device)
+    mlm_mask = mlm_ones * mlm_mask
+    mlm_mask = torch.transpose(mlm_mask, 1, 0)
+    # mlm_mask shuld be [batch, sequence, sequence]
+    mask = ((lm_mask + mlm_mask) > 0).float()
+    # mask shuld be [batch, sequence, sequence]
+    return mask
+
+
+@add_start_docstrings(
+    "The bare GPTSAN_JAPANESE Model transformer outputting logits.",
+    GPTSAN_JAPANESE_START_DOCSTRING,
+)
+class GPTSANJapaneseForConditionalGeneration(GPTSANJapanesePreTrainedModel):
```

Comment:
You should first define a `GPTSANJapaneseModel`, and then this class. Follow what was done in `SwitchTransformers`. This allows for better clarity and follows our usual architecture